sunpy/map/mapbase.py

```diff
@@ -1450,19 +1456,14 @@ def submap(self, bottom_left, top_right=None):
             [-95.92475   ,   6.028058  ,  -4.9797    ,  -1.0483578 ,
                 -3.9313421 ]], dtype=float32)
         """
+        if not isinstance(bottom_left, u.Quantity):
+            bottom_left, top_right = get_rectangle_coordinates(bottom_left,
```

Comment:
As @wafels points out I think we need to convert to the native map frame here?
```suggestion
            bottom_left = bottom_left.transform_to(self.coordinate_frame)
            bottom_left, top_right = get_rectangle_coordinates(bottom_left,
```

Reply:
I suggest reverting this and leaving it to a future PR.  This may open a can of worms, and is out of the scope of just making the APIs consistent.

Reply:
Yes this should be reverted it's my bad. The coordinate conversion is already handled in `world_to_pixel`.